clippy_lints/src/methods/mod.rs

```diff
@@ -3378,6 +3379,35 @@ declare_clippy_lint! {
     "calling `Stdin::read_line`, then trying to parse it without first trimming"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `bool::then` in `Iterator::filter_map`.
+    ///
+    /// ### Why is this bad?
+    /// This can be more concisely written with `filter` then `map` instead.
```

Comment:
Minor nit: 'concise' is not the right description. It reduces nesting and has a small readability gain due to separating the filtering step from the transformation step.

Otherwise this looks fine.